source/agora/node/FullNode.d

```diff
@@ -541,21 +475,14 @@ public class Node : API
         return this.enroll_man.hasEnrollment(enroll_hash);
     }
 
-    /// PUT: /receive_preimage
-    public void receivePreimage (PreImageInfo preimage) @safe
+    ///
+    protected void callReceivePreimage (PreImageInfo preimage) @safe
     {
-        log.trace("Received Preimage: {}", prettify(preimage));
-
-        if (this.enroll_man.addPreimage(preimage))
-            this.network.sendPreimage(preimage);
     }
 
-    /// GET: /get_preimage
-    public PreImageInfo getPreimage (Hash enroll_key)
+    ///
+    protected void callTryNominate () @safe
     {
-        PreImageInfo preimage;
-        this.enroll_man.getValidatorPreimage(enroll_key, preimage);
-        return preimage;
     }
```

Comment:
This `callTryNominate` looks suspicious.

Reply:
Minor changes and names have been changed.